Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/springcloud/SpringCloudAppNode.java

```diff
@@ -98,48 +119,50 @@ public String getSubscriptionId() {
-                new NodeActionListener() {
-                    @Override
-                    protected void actionPerformed(NodeActionEvent e) throws AzureCmdException {
-                        openResourcesInPortal(getSubscriptionId(), getAppId());
-                    }
-                }));
-        addAction(ACTION_OPEN_IN_BROWSER, new WrappedTelemetryNodeActionListener(SPRING_CLOUD, OPEN_IN_BROWSER_SPRING_CLOUD_APP,
-                new NodeActionListener() {
-                    @Override
-                    protected void actionPerformed(NodeActionEvent e) {
-                        if (StringUtils.isNotEmpty(app.properties().url())) {
-                            DefaultLoader.getUIHelper().openInBrowser(app.properties().url());
-                        } else {
-                            DefaultLoader.getUIHelper().showInfo(SpringCloudAppNode.this, "Public url is not available for app: " + app.name());
-                        }
-                    }
-                }));
+            new NodeActionListener() {
+                @Override
+                protected void actionPerformed(NodeActionEvent e) throws AzureCmdException {
+                    openResourcesInPortal(getSubscriptionId(), getAppId());
+                }
+            }));
+        addAction(ACTION_OPEN_IN_BROWSER, new WrappedTelemetryNodeActionListener(SPRING_CLOUD, OPEN_IN_BROWSER_SPRING_CLOUD_APP, new NodeActionListener() {
+            @Override
+            protected void actionPerformed(NodeActionEvent e) {
+                if (StringUtils.isNotEmpty(app.properties().url())) {
+                    DefaultLoader.getUIHelper().openInBrowser(app.properties().url());
+                } else {
+                    DefaultLoader.getUIHelper().showInfo(SpringCloudAppNode.this,
+                                                         "Public url is not available for app: "
+                                                                 + app.name());
+                }
+            }
+        }));
         addAction(ACTION_SHOW_PROPERTY, new WrappedTelemetryNodeActionListener(SPRING_CLOUD, SHOWPROP_SPRING_CLOUD_APP,
-                new NodeActionListener() {
-                    @Override
-                    protected void actionPerformed(NodeActionEvent e) {
-                        DefaultLoader.getUIHelper().openSpringCloudAppPropertyView(SpringCloudAppNode.this);
-                    }
-                }));
+            new NodeActionListener() {
+                @Override
+                protected void actionPerformed(NodeActionEvent e) {
+                    DefaultLoader.getUIHelper().openSpringCloudAppPropertyView(
+                           SpringCloudAppNode.this);
+                    // add this statement for false updating notice to update Property view
+                    // immediately
+                    SpringCloudStateManager.INSTANCE.notifySpringAppUpdate(clusterId,
```

Comment:
So we want to update the spring app status when open property view, why we use a force get method when get the spring app status?

Reply:
We want to use the existing spring app model for initiating spring cloud property view(replacing the http request to azure), so here we send a update notice with the cached data, in future, we will manage a cache towards azure resources, spring cloud property view will change to read from cache.